File: 1315. Sum of Nodes with Even-Valued Grandparent.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumEvenGrandparent = function(root) {
    let result = 0;
    const stack = [[root, null]];
    while(stack.length) {
        const [node, parent] = stack.shift();
        if (node) {
            if (node.left) {
                stack.push([node.left, node]);
                if (parent && parent.val %2 === 0) {
                    result += node.left.val;
                }
            } 
            if (node.right) {
                stack.push([node.right, node]);
                if (parent && parent.val %2 === 0) {
                    result += node.right.val;
                }
            } 
        }
    }
    return result;
};

File: Golang/Tree/1315.Sum-of-nodes-with-even-valued-grandparent.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func sumEvenGrandparent(root *TreeNode) int {
	ret := 0
	stack := []*TreeNode{root}
	check := func(node *TreeNode, grandparent int) {
		if grandparent%2 == 0 {
			if node.Left != nil {
				ret += node.Left.Val
			}
			if node.Right != nil {
				ret += node.Right.Val
			}
		}
	}
	for len(stack) > 0 {
		node := stack[len(stack)-1]
		stack = stack[:len(stack)-1]
		if node.Left != nil {
			check(node.Left, node.Val)
			stack = append(stack, node.Left)
		}
		if node.Right != nil {
			check(node.Right, node.Val)
			stack = append(stack, node.Right)
		}
	}
	return ret
}
